UniversalAlgebra/Algebras/Setoid.lagda

```diff
@@ -45,38 +48,126 @@ First we define an operator that translates an ordinary signature into a signatu
+ EqArgs refl u v = ∀ i → (_≈_ ξ) (u i) (v i)
 
-⟦ 𝑆 ⟧s ξ .isEquivalence .IsEquivalence.refl                        = refl , λ _ → Setoid.refl  ξ
-⟦ 𝑆 ⟧s ξ .isEquivalence .IsEquivalence.sym   (refl , g)            = refl , λ i → Setoid.sym   ξ (g i)
-⟦ 𝑆 ⟧s ξ .isEquivalence .IsEquivalence.trans (refl , g) (refl , h) = refl , λ i → Setoid.trans ξ (g i) (h i)
+IsEquivalence.refl  (isEqv (⟦ 𝑆 ⟧s ξ))                     = refl , λ _ → reflS  ξ
+IsEquivalence.sym   (isEqv (⟦ 𝑆 ⟧s ξ))(refl , g)           = refl , λ i → symS   ξ (g i)
+IsEquivalence.trans (isEqv (⟦ 𝑆 ⟧s ξ))(refl , g)(refl , h) = refl , λ i → transS ξ (g i) (h i)
 
 \end{code}
 
 
-##### Setoid Algebra
+##### Setoid Algebras
 
 A setoid algebra is just like an algebra but we require that all basic operations of the algebra respect the underlying setoid equality.
 The `Func` record packs a function (apply) with a proof (cong) that the function respects equality.
 
 \begin{code}
 
-Algebroid : (α ρ : Level)(𝑆 : Signature 𝓞 𝓥) → Type (𝓞 ⊔ 𝓥 ⊔ lsuc (α ⊔ ρ))
-Algebroid α ρ 𝑆 = Σ[ A ∈ Setoid α ρ ]      -- the domain (a setoid)
-                   Func (⟦ 𝑆 ⟧s A) A       -- the basic operations, along with proofs that each respects setoid equality
+Algebroid : (α ρ : Level) → Type (𝓞 ⊔ 𝓥 ⊔ lsuc (α ⊔ ρ))
+Algebroid α ρ = Σ[ A ∈ Setoid α ρ ]      -- the domain (a setoid)
+                 Func (⟦ 𝑆 ⟧s A) A       -- the basic operations,
+                                           -- along with congruence proofs that
+                                           -- each operation espects setoid equality
 
 record SetoidAlgebra α ρ : Type (𝓞 ⊔ 𝓥 ⊔ lsuc (α ⊔ ρ)) where
   field
-    Den  :  Setoid α ρ
-    den  :  Func (⟦ 𝑆 ⟧s Den) Den
+    Domain : Setoid α ρ
+    Interp : Func (⟦ 𝑆 ⟧s Domain) Domain
      --      ^^^^^^^^^^^^^^^^^^^^^^^ is a record type with two fields:
      --       1. a function  f : (⟦ 𝑆 ⟧s Den) .Carrier  → Den . Carrier
      --       2. a proof cong : f Preserves _≈₁_ ⟶ _≈₂_ (that f preserves the setoid equalities)
 
 \end{code}
 
+Easier notation for application of an (interpreted) operation symbol.
+
+\begin{code}
+
+_̂_ : {α ρ : Level} (f : ∣ 𝑆 ∣)(𝑨 : Algebroid α ρ) → (∥ 𝑆 ∥ f  →  Carrier ∣ 𝑨 ∣) → Carrier ∣ 𝑨 ∣
+
+f ̂ 𝑨 = λ a → apply ∥ 𝑨 ∥ (f , a)
+
+\end{code}
+
+\end{code}
+
+#### Products of Algebroids
+
+\begin{code}
+
+open Func           using    ( cong                     )
+                    renaming ( f             to  apply  )
+open Setoid         using    ( Carrier       ;   _≈_    )
+                    renaming ( isEquivalence to  isEqv  )
+open IsEquivalence  renaming ( refl          to  reflE
+                             ; sym           to  symE
+                             ; trans         to  transE )
+
+module _ {α ρ ι : Level} where
+
+ ⨅ : {I : Type ι }(𝒜 : I → Algebroid α ρ) → Algebroid (α ⊔ ι) (ρ ⊔ ι)
+
+ ⨅ {I} 𝒜 = domain , interp-ops
+  where
+  domain : Setoid _ _
+  domain = record { Carrier = ∀ i → Carrier ∣ 𝒜 i ∣
```

Comment:
I'm quite sure there are combinators in Relation.*ary.Indexed.* that would make this look much nicer. My knowledge of Agda has benefited a lot from just reading fairly large parts of stdlib.  I start from https://agda.github.io/agda-stdlib/ and just click around!